lib/src/kbasesearchengine/system/TypeFileStorage.java

```diff
@@ -159,6 +163,8 @@ public TypeFileStorage(
         Utils.nonNull(fileLister, "fileLister");
         Utils.nonNull(logger, "logger");
         storageTypes = processTypesDir(typesDir, searchSpecParser, fileLister, logger);
+
+        //not sure what the purpose of this is. We could just pair the types together based on storage type
```

Comment:
If 'this' means the type mappings, then the purpose of the type mappings is to map specific versions of a storage type against specific versions of the search type. Otherwise there's no way to determine which version of a search type to pick given a storage type. Recall that storage type versions are not equivalent across environments.